discovery-provider/src/tasks/users.py

```diff
@@ -79,6 +77,83 @@ def user_state_update(
                             txhash,
                         )
                         user_events_lookup[user_id] = {"user": ret_user, "events": []}
+                except Exception as e:
+                    logger.error("Error in parse user transaction")
+                    event_blockhash = update_task.web3.toHex(block_hash)
+                    raise IndexingError(
+                        "user", block_number, event_blockhash, txhash, str(e)
+                    ) from e
+
+    creator_node_endpoint = {
+        user_id: data["user"].creator_node_endpoint
+        for user_id, data in user_events_lookup.items()
+    }
+    cids = []
+    for tx_receipt in user_factory_txs:
+        txhash = update_task.web3.toHex(tx_receipt.transactionHash)
+        for event_type in user_event_types_arr:
+            user_events_tx = getattr(user_contract.events, event_type)().processReceipt(
+                tx_receipt
+            )
+            for entry in user_events_tx:
+                user_id = entry["args"]._userId
+                if event_type == user_event_types_lookup["update_multihash"]:
+                    metadata_multihash = helpers.multihash_digest_to_cid(
+                        entry["args"]._multihashDigest
+                    )
+                    is_blacklisted = is_blacklisted_ipld(session, metadata_multihash)
+                    if not is_blacklisted:
+                        cids.append(
+                            [metadata_multihash, creator_node_endpoint[user_id]]
+                        )
+                elif (
```

Comment:
this depends on the answer to why we need creator_node_endpoint but i'm not sure why we need this elif here